Source/WTF/wtf/CompletionHandler.h

```diff
@@ -96,7 +112,8 @@ class CompletionHandlerWithFinalizer<Out(In...)> {
 
     template<typename CallableType, class = typename std::enable_if<std::is_rvalue_reference<CallableType&&>::value>::type>
     CompletionHandlerWithFinalizer(CallableType&& callable, Function<void(Function<Out(In...)>&)>&& finalizer)
-        : m_function(std::forward<CallableType>(callable))
+        : m_callThread(threadAssertionFromCallable(callable).value_or(ThreadLikeAssertion { }))
```

Comment:
You can observe above that the CompletionHandlerWithFinalizer is implemented incorrectly, so you should fix it.
The type should take in two callable, and guarantee that either one is called or the other.

Each callable should ensure that if they are  called, they called in their expected threadlikes.